wip-trampoline.md

```diff
@@ -229,33 +229,48 @@ TODO: use a TLV stream for `node_update`.
 TODO: should use an odd type (261?) because it's safe to ignore if unsupported?
 
 It has a structure similar to the `channel_update` message but is channel
-agnostic. Trampoline nodes need to estimate a `cltv_expiry_delta` and `fees`
-that would allow them to route to any other trampoline node. `Node_update`s
-should be relayed the same way `channel_update`s are relayed.
+agnostic. `Node_update`s should be relayed the same way `channel_update`s are
+relayed (staggered broadcast).
+
+If a node is not willing to relay trampoline payments, it will simply never
+send a `node_update`. This lets the network discriminate nodes that are able to
+relay trampoline payments from nodes that use it but simply as sender or
+destination.
 
 ### Fees and CLTV estimation
 
-TODO: detail heuristics that can be used by trampoline nodes to estimate cltv
-and fees.
+Trampoline nodes need to estimate a `cltv_expiry_delta` and `fee` that would
+allow them to route to any other trampoline node while being competitive with
+other nodes. This is a great opportunity to incentivize nodes to open channels
+between each other to minimize the cost of trampoline hops. This is also a
+great opportunity for nodes to implement smart fee estimation algorithms as
+a competitive advantage.
```

Comment:
Is it? I would expect the trampoline to just do trial-and-error with the maximum fee allocated to themselves just like the original sender would. That's not really estimating fees at all. They even get told how much funds MUST arrive at the next trampoline.

Reply:
> They even get told how much funds MUST arrive at the next trampoline.

But that will often be based on the fee they advertised (via their estimation algorithm) ;)

> I would expect the trampoline to just do trial-and-error

They could, but there's a risk they will end up locking up funds if one of the nodes in the route is unresponsive and they have to wait until the timeout. It's hard to predict because I think the heuristics will depend on the reliability of the overall network, but I think both options could make sense.